File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        def swap(red, blue, i):
            if(i >= red and nums[i] == 0):
                nums[red], nums[i] = nums[i], nums[red]

            if(i <= blue and nums[i] == 2):
                nums[blue], nums[i] = nums[i], nums[blue]
        
        n = len(nums)
        i = red = 0
        blue = n - 1
        
        while(i < n):
            while(red < n and nums[red] == 0):
                red += 1
            
            while(blue > 0 and nums[blue] == 2):
                blue -= 1
            
            swap(red, blue, i)
            swap(red, blue, i)
            i += 1
        
        

File: 0075-sort-colors/0075-sort-colors.go
func sortColors(nums []int)  {
    var c0, c1, c2 = 0, 0, 0
    for _, num := range nums {
        if num == 0 {
            c0++
        } else if num == 1 {
            c1++
        } else {
            c2++
        }
    }
    
    for i := range len(nums){
        if c0 > 0 {
            nums[i] = 0
            c0--
        } else if c1 > 0 {
            nums[i] = 1
            c1--
        } else {
            nums[i] = 2
        }
    }
}